libdnf/module/module_sack.cpp

```diff
@@ -478,6 +482,149 @@ std::pair<std::vector<std::vector<std::string>>, ModuleSack::ModuleErrorType> Mo
 }
 
 
+static std::pair<std::string, std::string> parse_platform_id_from_string(const std::string & platform_id) {
+    auto index = platform_id.find(':');
+    if (index == std::string::npos) {
+        return {};
+    } else {
+        return make_pair(platform_id.substr(0, index), platform_id.substr(index + 1));
+    }
+}
+
+
+static std::pair<std::string, std::string> parse_platform_id_from_file(const std::string & os_release_path) {
```

Comment:
May I ask you why the function throw runtime_error? Anyway it is always catch and logged and then the code continue?

Reply:
It was mentioned to terminate on any problem when reading the `os-release` file and then just log the message with the error, because we could have more files from where the platform ID could be detected, so we continue with another attempts. This is basically taken over from `DNF` and I am open for discussion for any better solution. Or is there some specific problem with this approach?

Reply:
No, there is no a problem with the approach. When I see that construct I always try to understand details. In the function there is an alternative source of exceptions `libdnf::utils::fs::File file{os_release_path, "r"};`, which will require catch anyway. May be could you add a short description of the method that it raises not only `std::runtime_error` but also `std::filesystem::filesystem_error`? Do we expect any other type of error?

Reply:
I've changed the exception for parsing problems to the custom one. I don't know about adding the description about internal raising of the filesystem exception. I don't see this approach used in other classes and also this is just a static function in the `.cpp`. But of course if you find it useful, I will add it.

Reply:
It would be better to describe what kind of exception the method could raise. It will make life easier for future internal users.

Reply:
Docs were updated.